adagrams/game.py

```diff
@@ -1,13 +1,148 @@
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 
 def draw_letters():
-    pass
+    letter_bank = []
+    available_letters = build_letters_and_availability_list(LETTER_POOL)
+    
+    while len(letter_bank) < 10:
+        index_of_letter = get_index_of_letter()
 
+        if not available_letters[index_of_letter]['is_not_available']:
+            letter_bank.append(available_letters[index_of_letter]['letter'])
+            available_letters[index_of_letter]['is_not_available'] = True
+
+    reset_available_letters(available_letters)
+
+    return letter_bank
+
+def build_letters_and_availability_list(letter_pool):
+    available_letters = []
+    
+    for letter, letter_distribution in letter_pool.items():
+        for single_letter in range(letter_distribution):
+            available_letters.append({'letter':letter, 'is_not_available':False})
+    
+    return available_letters
+     
 def uses_available_letters(word, letter_bank):
-    pass
+    letters_in_word = list(word.upper())
+    is_valid = False
+
+    for letter in word:
+        same_letter_count = letters_in_word.count(letter) <= letter_bank.count(letter)
+        
+        if word.count(letter) > 0 and same_letter_count:
+            continue
+        else:
+            return is_valid
+    
+    is_valid = True
+
+    return is_valid
 
 def score_word(word):
-    pass
+    SCORE_CHART = {
+        1 : ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+        2 : ['D', 'G'],
+        3 : ['B', 'C', 'M', 'P'],
+        4 : ['F', 'H', 'V', 'W', 'Y' ],
+        5 : ['K'],
+        8 : ['J', 'X' ],
+        10 : ['Q', 'Z']
+    }
+    BONUS_POINTS = 8
+    score = 0
+    word_case_insensitive = word.upper()
+
+    for letter in word_case_insensitive:
+        for points, letters in SCORE_CHART.items():
+            if letter in letters:
+                score += points
+
+    if len(word) >= 7:
+        score += BONUS_POINTS
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    words_and_scores = build_words_and_scores_dict(word_list)
+    top_score = 0
+    highest_scores = []
+    winning_word_and_score = []
+
+    for current_word, current_score in words_and_scores.items():
+        if top_score == 0 or current_score == top_score:
+            top_score = current_score
+            highest_scores.append([current_word, current_score])
+        elif current_score > top_score:
+            top_score = current_score
+            highest_scores[0] = [current_word, current_score]
+    
+    if len(highest_scores) > 1:
+        winning_word_and_score = check_for_a_tie(highest_scores)
+    else:
+        winning_word_and_score = highest_scores[0]
+
+    return tuple(winning_word_and_score)
+
+def get_index_of_letter():
```

Comment:
Nice helper! Your could code be even more self-documenting if this name also included the randomness factor maybe like `get_rand_index`